superset-frontend/src/explore/components/controls/FilterControl/AdhocFilterEditPopoverSimpleTabContent/index.jsx

```diff
@@ -236,18 +237,42 @@ export default class AdhocFilterEditPopoverSimpleTabContent extends React.Compon
   }
 
   isOperatorRelevant(operator, subject) {
+    const dataSourceType = this.props.datasource.type;
+    const column = this.props.datasource.columns.find(
+      col => col.column_name === subject,
+    );
+    const isColumnBoolean =
+      !!column && (column.type === 'BOOL' || column.type === 'BOOLEAN');
+    const isColumnNumber = !!column && column.type === 'INT';
+
     if (operator && CUSTOM_OPERATORS.has(operator)) {
       const { partitionColumn } = this.props;
       return partitionColumn && subject && subject === partitionColumn;
     }
-
+    if (isColumnBoolean) {
+      return (
+        [
+          OPERATORS['IS NULL'],
+          OPERATORS['IS NOT NULL'],
+          ...BOOLEAN_ONLY_OPERATORS,
+        ].indexOf(operator) >= 0
+      );
+    }
+    if (dataSourceType === 'druid' && !isColumnNumber) {
+      return (
+        [...BOOLEAN_ONLY_OPERATORS, ...TABLE_ONLY_OPERATORS].indexOf(operator) <
+        0
+      );
+    }
+    if (dataSourceType === 'table' && !isColumnNumber) {
```

Comment:
For calculated column, which could be number, string, date, or unset type, we probably show IS_TRUE, IS_FALSE as option?